### PR TITLE
vscode-extensions.ms-vscode.PowerShell: init at 2022.7.2

### PR DESCRIPTION
**Pull request by rhoriguchi**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by superherointj** [comment]
@rhoriguchi Can you complete meta information?
Add download page as marketplace page.
Also, consider adding yourself as maintainer.

**Comment by rhoriguchi** [comment]
> @rhoriguchi Can you complete meta information? Add download page as marketplace page (https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell). Also, consider adding yourself as maintainer.

As requested I've updated the metadata and added myself as a maintainer.

**Comment by superherointj** [comment]
@rhoriguchi The package is erroring as:
`error: undefined variable 'maintainers'`
Likely, you have forgotten `with lib;` in meta.
Certify to build the package after every change.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -1674,6 +1674,22 @@ let
...
       ms-vscode.cpptools = callPackage ./cpptools { };
 
+      ms-vscode.PowerShell = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "PowerShell";
+          publisher = "ms-vscode";
+          version = "2022.6.3";
+          sha256 = "sha256-P8kakmpT3yAkKFroKttiLOV7PneOIoQOEtwImPIDHQY=";
+        };
+        meta = {
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1684]:
```suggestion
        meta = with lib; {
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -1674,6 +1674,22 @@ let
...
+          description = "PowerShell Language Support for Visual Studio Code.";
+          downloadPage = "https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell";
+          homepage = "https://github.com/PowerShell/vscode-powershell";
+          license = lib.licenses.mit;
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1688]:
```suggestion
          license = licenses.mit;
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -1674,6 +1674,22 @@ let
...
 
+      ms-vscode.PowerShell = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "PowerShell";
+          publisher = "ms-vscode";
+          version = "2022.6.3";
+          sha256 = "sha256-P8kakmpT3yAkKFroKttiLOV7PneOIoQOEtwImPIDHQY=";
+        };
+        meta = {
+          description = "PowerShell Language Support for Visual Studio Code.";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1685]:
```suggestion
          description = "PowerShell Language Support for Visual Studio Code";
```

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:1685]: Please remove "." from description to comply with guidelines:

https://nixos.org/manual/nixpkgs/stable/#var-meta-description

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:1685]: Description could be improved too:
`A Visual Studio Code extension for PowerShell language support` (feel free to improve it further)

**Comment by rhoriguchi** [comment]
@superherointj sorry for the hassle and thx for the reviews.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 182817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.ms-vscode.PowerShell</li>
  </ul>
</details>

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -1674,6 +1674,22 @@ let
 
       ms-vscode.cpptools = callPackage ./cpptools { };
 
+      ms-vscode.PowerShell = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "PowerShell";
+          publisher = "ms-vscode";
+          version = "2022.6.3";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1681]:
```suggestion
          version = "2022.7.2";
```

**Comment by superherointj** [comment]
@rhoriguchi A new version is out. Can you update?

**Comment by rhoriguchi** [comment]
> @rhoriguchi A new version is out. Can you update?

Is updated :)

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 182817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.ms-vscode.PowerShell</li>
  </ul>
</details>

**Review by superherointj — COMMENTED** [review]
LGTM

**Pull request merged**